src/proto/streams/streams.rs

```diff
@@ -172,9 +172,19 @@ where
 
         let id = frame.stream_id();
 
+        if me.store.is_reset(&id) {
```

Comment:
As mentioned above, this doubles all hash lookups for received strams.

Also, I think that the stream window flow should still be validated? We just drop frames that the remote sent before seeing the RST_STREAM, we still want to ensure that the connection state is sane.

Reply:
If we do keep a second smaller map, then I can do some stupid syntax crap to fix the borrow checker. Since a mutable stream is taken from the store, even the `None` case can't access the store again...

Reply:
Regarding counting the stream window as well: since the stream is reset, the window should be closed.

> Flow-controlled frames (i.e., DATA) received after sending RST_STREAM are counted toward the connection flow-control window.

Reply:
Right, it needs to be counted against the connection flow control, but unless we keep the stream level flow state around too, we don't know if the received data was a valid size (exceeded the max size the stream would have accepted if it wasn't locally reset).

Basically, we can't differentiate between the remote peer being correct and just not having seen the RST_STREAM vs. the remote being buggy and violating flow control.